### PR TITLE
[Enhancement] Support unnest table function for trino parser

### PR DESCRIPTION
**Pull request by Youngwb**
Fixes #issue
#14825 
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 29 / 29 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/parser/trino/AstBuilder.java|29|29|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java [hunk 1]
```diff
@@ -396,6 +398,48 @@ protected ParseNode visitValues(Values node, ParseTreeContext context) {
...
+    @Override
+    protected ParseNode visitUnnest(Unnest node, ParseTreeContext context) {
+        List<Expr> arguments = visit(node.getExpressions(), context, Expr.class);
+        List<Expr> expressions = new ArrayList<>();
+
+        for (Expr arg : arguments) {
+            if (arg instanceof ArrayExpr) {
+                if (!arg.getChildren().isEmpty()) {
+                    // need to covert array[row(1,2), row(3,4))] to array[1,3], array[2,4], so SR could unnest it
+                    Expr firstArrayElement = arg.getChildren().get(0);
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java:410]: We only support `array[row(literals)]` this format?

**Review by dirtysalt — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java:410]: or array[literals]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**